src/main/java/org/littleshoot/proxy/impl/ProxyToServerConnection.java

```diff
@@ -528,6 +531,8 @@ private void respondWith(HttpObject httpObject) {
     private void connectAndWrite(HttpRequest initialRequest) {
         LOG.debug("Starting new connection to: {}", remoteAddress);
 
+        this.clientConnection.channel.attr(remoteAddressAttrKey).set(remoteAddress);
+
```

Comment:
do we need to remove this once the connection has completed? otherwise we're going to maintain a reference to these channels forever won't we?